### DISCUSSION TITLE
Modify header field or subject

### DISCUSSION THREAD

**Discussion by caco3** [mjs/imapclient#468] [Q&A] [answered]
Is there a way to modify the subject or add/modify a header field?

I am building a spam filter and need to add the score somehow to an email.

Eg (Header):
```
X-Spam-Score: 20
```
or, if this is not possible, it could be added to the subject:
```
***** SPAM ***** (10.8) ***** Newsletter abcd
```

**Comment by mjs** [comment, marked as answer]
I'm afraid this isn't really possible. IMAP messages are basically immutable.

The best you can do is download the entire message, modify the headers and append it to the mailbox, and then delete the original one.

This Stack Overflow questions covers the topic: https://stackoverflow.com/questions/16945041/imap-editing-a-draft-or-existing-message

**Reply by caco3** [reply]
Thanks for the explanation.
I actually expected something like this :(

**Comment by mjs** [comment]
With the `append` method: https://imapclient.readthedocs.io/en/2.2.0/api.html#imapclient.IMAPClient.append

`APPEND` is the IMAPClient command for adding a message to a mailbox.
